src/algebraic_geometry/presheafed_space/gluing.lean

```diff
@@ -0,0 +1,434 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import topology.gluing
+import algebraic_geometry.open_immersion
+import algebraic_geometry.locally_ringed_space.has_colimits
+
+/-!
+# Gluing Structured spaces
```

Comment:
Is a "structured space" just a presheafed space?

Reply:
probably also sheafed spaces, locally ringed spaces and schemes in a later PR.
Though I'm still not sure whether they should all be in one file.
If so we might want to move it out from the `preheated_space` folder?